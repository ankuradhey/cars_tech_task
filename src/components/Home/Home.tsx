import React, { FC, useState } from "react";
import { Container, Row, Col } from "react-bootstrap";
import { FilterState } from "../../types";
import { FilterResult } from "../FilterResult";
import { Sidebar } from "../Sidebar";

export const Home: FC = () => {
    const [filter, setFilter] = useState<FilterState>({ color: "", manufacturer: "" });

    const submitFilter = (val) => {
        setFilter({ ...val });
    };

    return (
        <Container>
            <Row>
                <Col md={4}>
                    <Sidebar setFilter={submitFilter} />
                </Col>
                <Col md={8}>
                    <FilterResult
                        color={filter?.color ?? ""}
                        manufacturer={filter?.manufacturer ?? ""}
                    />
                </Col>
            </Row>
        </Container>
    );
};
