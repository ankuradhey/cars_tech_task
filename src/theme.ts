import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
