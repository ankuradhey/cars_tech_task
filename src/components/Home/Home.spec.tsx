import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";
import { Home } from "./Home";

describe("Home Component", () => {
    it("Should render Home component", () => {
        const history = createMemoryHistory();
        const route = "/";
        history.push(route);

        const { container, getByText } = render(
            <ThemeProvider theme={defaultTheme}>
                <Router history={history} initialEntries={["/"]}>
                    <Home />
                </Router>
            </ThemeProvider>
        );
        expect(getByText("Color")).toBeInTheDocument();
        expect(getByText("Manufacturer")).toBeInTheDocument();
        expect(getByText("Available cars")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });
});
