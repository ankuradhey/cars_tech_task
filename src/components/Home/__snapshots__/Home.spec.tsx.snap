// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render Home component 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-md-4"
    >
      <div
        class="margin-bottom-3 card"
      >
        <div
          class="card-body"
        >
          <form
            class=""
          >
            <div
              class="form-group"
            >
              <label
                class="form-label"
              >
                <div
                  class="sc-dlfnbm icINku"
                >
                  Color
                </div>
              </label>
              <select
                class="custom-select"
                data-testid="color"
              >
                <option
                  value=""
                >
                  All car colors
                </option>
              </select>
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
              >
                <div
                  class="sc-dlfnbm icINku"
                >
                  Manufacturer
                </div>
              </label>
              <select
                class="custom-select"
                data-testid="manufacturer"
              >
                <option
                  value=""
                >
                  All manufacturers
                </option>
              </select>
            </div>
          </form>
          <button
            class="sc-gKsewC euEXZP float-right"
            data-testid="filter"
          >
            Filter
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-md-8"
    >
      <div
        class="sc-dlfnbm iPsSGC d-block margin-bottom-2"
      >
        Available cars
      </div>
      <div
        class="sc-dlfnbm dqWGDI margin-bottom-3"
      >
        Loading
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ bAWNIY"
              height="80px"
              loading="true"
              width="100px"
            />
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="sc-bdfBwQ ghPXQo margin-bottom-1"
              height="18px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ gTqLku margin-bottom-2"
              height="12px"
              loading="true"
              width="100%"
            />
            <div
              class="sc-bdfBwQ kAwMCN margin-bottom-2 d-inline-block"
              height="12px"
              loading="true"
              width="auto"
            >
              <button
                class="sc-gsTCUz ezYRfg invisible"
              >
                View details
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-flex justify-content-center"
      >
        <div
          class="p-2 bd-highlight"
        >
          <div
            class="sc-dlfnbm icINku"
            data-testid="first-page"
          >
            First
          </div>
        </div>
        <div
          class="p-2 bd-highlight"
        >
          <div
            class="sc-dlfnbm icINku"
            data-testid="previous-page"
          >
            Previous
          </div>
        </div>
        <div
          class="p-2 bd-highlight"
        >
          <div
            class="sc-dlfnbm icINku"
          >
            Page 
            1
             of 
          </div>
        </div>
        <div
          class="p-2 bd-highlight"
        >
          <div
            class="sc-dlfnbm icINku"
            data-testid="next-page"
          >
            Next
          </div>
        </div>
        <div
          class="p-2 bd-highlight"
        >
          <div
            class="sc-dlfnbm icINku"
            data-testid="last-page"
          >
            Last
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
